## Practical session II –  Introduction to RNA-seq Analysis  
## Instructor: Ravza Gür

### Tutorial link
- https://rnabio.org/
- https://github.com/griffithlab/rnaseq_tutorial

### Tutorial paper
[Informatics for RNA-seq: A web resource for analysis on the cloud](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004393)

### Test data
[HBR_Rep1_chr22](http://genomedata.org/rnaseq-tutorial/HBR_UHR_ERCC_ds_5pc.tar)


### Package list
- [FastQC - info](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [FastQC - conda installation](https://anaconda.org/bioconda/fastqc)
- [STAR Aligner Manual - info](https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf)
- [STAR Aligner - conda installation](https://anaconda.org/bioconda/star)
- [Samtools - info](http://www.htslib.org/)
- [Samtools - conda installation](https://anaconda.org/bioconda/samtools)
- [FeatureCounts from Subread - info](http://bioinf.wehi.edu.au/featureCounts/)
- [Subread - conda installation](https://anaconda.org/bioconda/subread)
- [Integrative Genomics Viewer - IGV](http://software.broadinstitute.org/software/igv/home)


### Some Reading
- [A survey of best practices for RNA-seq data analysis](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0881-8)
- [How many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use?](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4878611/)
- [Sequence Alignment/Map Format Specification](https://samtools.github.io/hts-specs/SAMv1.pdf)
- [Current best practices in single-cell RNA-seq analysis: a tutorial](https://www.embopress.org/doi/full/10.15252/msb.20188746)

### Further info
- [Bioconductor - Open Source Software for Bionformatics](https://www.bioconductor.org/)
- [Gene Expression Omnibus Database - GEO](https://www.ncbi.nlm.nih.gov/sites/GDSbrowser/)
